src/parser.c

```diff
@@ -0,0 +1,108 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "parser.h"
+
+#define MAXCHAR 1025	// limit is 1024
+#define INITLINE 200
+#define READ_BUFFERSIZE 256
+#define LSH_TOK_BUFFERSIZE 64
+#define LSH_TOK_DELIM " \t\r\n\a"
+
+/* Parsing Functions for Parsing Input Files */
+
+/* returns with pointer to array of strings, each of which represent a line in a given file */
+/* current limit to each string is 1000 characters */
+/* return NULL if file could not be opened */
+char** parse_file(char* filename)
+{
+	FILE *f = fopen(filename, "r");
+
+	if (f == NULL) {
+		return NULL;
+	}
+
+    // initialization
+	int n = 0;
+	int i = INITLINE;
+	char str[MAXCHAR];
+	char** lines;
+	lines = calloc(i, sizeof(char*));
+	if (lines == NULL) {
+		fprintf(stderr, "lineparse_file: malloc failed\n");
+		exit(0);
+	}
+
+    // read each line into a string. This size is currently limited at 1025 lines (1024, including terminating char).
+    // resize if necessary while reading file
+	while(fgets(str, MAXCHAR, f) != 0) {
+		lines[n] = strdup(str);
+		n++;
+	}
+	while(n < i) {  // initialize the rest of the lines, which do not hold new characters
+		lines[n] = NULL;
+		n++;
+	}
+
+	fclose(f);
+	return lines;
+}
+
+/* Parsing functions for parsing command line inputs */
+/* read a command line and return a string */
+char* parse_read_line()
+{
+	char input[READ_BUFFERSIZE];
+	char* rval;
+	memset(input, '\0', READ_BUFFERSIZE);
```

Comment:
This seems wildly unsafe if it's acting on undefined memory.

Reply:
malloced

Reply:
This is unnecessary; `calloc` zeroes memory.